core/src/test/scala/unit/kafka/server/AbstractApiVersionsRequestTest.scala

```diff
@@ -49,7 +49,9 @@ abstract class AbstractApiVersionsRequestTest(cluster: ClusterInstance) {
     val overrideHeader = IntegrationTestUtils.nextRequestHeader(ApiKeys.API_VERSIONS, Short.MaxValue)
     val socket = IntegrationTestUtils.connect(cluster.brokerSocketServers().asScala.head, cluster.clientListener())
     try {
-      IntegrationTestUtils.sendWithHeader(request, overrideHeader, socket)
+      val serializedBytes = Utils.toArray(
```

Comment:
Why this change is necessary? It seems to me the code gets more complicated.

Reply:
Because this method is used to test the version inconsistency between request header and request body, but `IntegrationTestUtils.sendWithHeader` will inspect the version and throws `IllegalArgumentException` on inconsistency after this PR.